sklearn/utils/estimator_checks.py

```diff
@@ -440,7 +474,7 @@ def check_estimator_sparse_data(name, estimator_orig):
                       "sparse input is not supported if this is not the case."
                       % name)
                 raise
-        except Exception:
+        except Exception as e:
```

Comment:
You should build a dummy estimator in `sklearn/utils/tests/test_estimator_checks.py` that causes this error to occur, to make sure the logic and validation is all working correctly.

Reply:
I will build a dummy estimator

plus you were right about scipy issue, its working fine with version 0.14 or above